youtube_dl/extractor/njpwworld.py

```diff
@@ -64,35 +71,28 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, video_id)
 
         formats = []
-        for mobj in re.finditer(r'<a[^>]+\bhref=(["\'])/player.+?[^>]*>', webpage):
-            player = extract_attributes(mobj.group(0))
-            player_path = player.get('href')
-            if not player_path:
-                continue
-            kind = self._search_regex(
-                r'(low|high)$', player.get('class') or '', 'kind',
-                default='low')
+        for kind, vid in re.findall(r'if\s+\(\s*imageQualityType\s*==\s*\'([^\']+)\'\s*\)\s*{\s*video_id\s*=\s*"(\d+)"', webpage):
+            player_path = '/intent?id=%s&type=url' % vid
             player_url = compat_urlparse.urljoin(url, player_path)
-            player_page = self._download_webpage(
-                player_url, video_id, note='Downloading player page')
-            entries = self._parse_html5_media_entries(
-                player_url, player_page, video_id, m3u8_id='hls-%s' % kind,
-                m3u8_entry_protocol='m3u8_native')
-            kind_formats = entries[0]['formats']
-            for f in kind_formats:
-                f['quality'] = 2 if kind == 'high' else 1
-            formats.extend(kind_formats)
+            formats.append({
+                'url': player_url,
+                'format_id': kind,
+                'ext': 'mp4',
+                'protocol': 'm3u8',
+                'quality': 2 if kind == 'high' else 1,
+            })
 
         self._sort_formats(formats)
 
-        post_content = get_element_by_class('post-content', webpage)
+        tag_block = get_element_by_class('tag-block', webpage)
+        tag_list = get_element_by_class('tag-list', tag_block) if tag_block else None
```

Comment:
why is this extracted in two steps now, and not directly from `tag-block`?

Reply:
Here's the HTML structure: Inside the `tag-block` div are the `tag-list` div (which has the `tag-` links) and a "More" link. This way was to exclude the "More" link in lieu of the "element class" check. Now we have a redundancy with both the `tag-list` and `tag-` filters.

Reply:
after the change to check the `a` HTML element `class` attribute, this is no longer needed.

Reply:
I already said it's a redundancy. It'd be clearer if you just said, "The `a` check is better than the `div` check. Go with that."